modules/tutorials/pages/java-tutorial/second-couchbase-application.adoc

```diff
@@ -0,0 +1,124 @@
+----
+include::partial$graphic-design-course.json[]
+----
+--
+
+fine art::
++
+--
+[source, json]
+----
+include::partial$fine-art-course.json[]
+----
+--
+====
+
+[NOTE]
+====
+The documents include an additional `type` field. Do you remember why we include a `type` for all our documents?
+
+.Answer
+[%collapsible]
+=====
+Documents of different types can reside in the same bucket/collection, so it's useful to include a field which can be used to filter documents by type.
+=====
+====
+
+The code should be familiar to you; there's not much difference between writing to the course collection and writing to the student collection; you just have more records to deal with:
+
+[source, java]
+----
+import com.couchbase.client.java.Bucket;
+import com.couchbase.client.java.Cluster;
+import com.couchbase.client.java.Collection;
+import com.couchbase.client.java.Scope;
+import com.couchbase.client.java.json.JsonObject;
+
+import java.time.Duration;
+
+public class InsertCourses {
+
+    public static void main(String[] args) {
+        Cluster cluster = Cluster.connect("localhost",
+                "Administrator", "password");
+
+        Bucket bucket = cluster.bucket("student-bucket");
+        bucket.waitUntilReady(Duration.ofSeconds(10));
+        Scope scope = bucket.scope("art-school-scope");
+        Collection course_records = scope.collection("course-record-collection");    // <.>
+
+        addCourse(course_records, "ART-HISTORY-000001", "art history", "fine art", 100);
+        addCourse(course_records, "FINE-ART-000002", "fine art", "fine art", 50);
+        addCourse(course_records, "GRAPHIC-DESIGN-000003", "graphic design", "media and communication", 200);
+
+        cluster.disconnect();
+    }
+
+    private static void addCourse(Collection collection, String id, String name,
+                                  String faculty, int creditPoints) {
+
+        JsonObject course = JsonObject.create()
+                .put("course-name", name)
+                .put("faculty", faculty)
+                .put("credit-points", creditPoints)
+                .put("type", "course");
+
+        collection.upsert(id, course);
+
+    }
+}
+----
+
+<.> Note that you're now writing to a different collection.
+
+IMPORTANT: Make sure that you've created the `course-collection` in the admin console before you attempt to run the program.
+
+You can use maven to run the application:
+
+[source, bash]
+----
+mvn exec:java -Dexec.mainClass="InsertCourses" -Dexec.cleanupDaemonThreads=false
+----
+
+Use the admin console to make sure the documents have been created in the correct collection.
+
+image::new-course-collection.png[]
+
+== Next steps
+
+So you've created a cluster, a bucket, a scope and two collections. You've also populated your collections with documents. Well, a database isn't much use until we can retrieve information from it, which is what you're going to take a look at in the xref:java-tutorial/retrieving-documents.adoc[next part].
```

Comment:
Do we now need this explicit link, as we have pagination?

Reply:
Not really, but without the link, it's a bit weird. I'll think about rewriting